src/Storage/FileSystem/FileSystemCacheAdapter.php

```diff
@@ -205,37 +245,63 @@ public function getMultiple($keys, $default = null)
      *                                      the driver supports TTL then the library may set a default value
      *                                      for it or let the driver take care of that.
      *
-     * @return null
+     * @return bool True on success and false on failure.
      *
-     * @throws PageCacheException
+     * @throws \PageCache\Storage\CacheAdapterException
+     * @throws \Psr\SimpleCache\InvalidArgumentException
      *   MUST be thrown if $values is neither an array nor a Traversable,
      *   or if any of the $values are not a legal value.
      */
     public function setMultiple($values, $ttl = null)
     {
-        throw new PageCacheException(__METHOD__.' not implemented');
+        if (!$values || (!\is_array($values) && !$values instanceof \Traversable)) {
+            throw new InvalidArgumentException('Cache values must be an array or Traversable');
+        }
+
+        $result = true;
+
+        foreach ($values as $key => $value) {
+            if (\is_int($key)) {
+                $key = (string)$key;
+            }
+
+            $result = $this->set($key, $value, $ttl) && $result;
```

Comment:
let's have `(string)$key` here, and remove 3 lines of `if` above.

Reply:
Sorry, due to the PSR-16 spec we need to throw an exception if `$values` array contains non-string keys, but integer keys must be treated as string ones. So this semi-ugly cast is needed to comply with PSR-16 spec.